pyne/ace.pyx

```diff
@@ -1555,10 +1735,121 @@ class Reaction(object):
+                h = fa - fb
+                # Calculate a[k], b[k], and slope terms
+                ak = y_inv_sq * h[2] + 2 * y_inv * h[1] + h[0]
+                bk = y_inv_sq * h[4] + 4 * y_inv * h[3] + 6 * h[2] + \
+                     4 * y * h[1] + y_sq * h[0]
+                slope = (xs[k + 1] - xs[k]) / (x[k + 1] ** 2 - x[k] ** 2)
+                # Add contribution to broadened cross section
+                sigma += ak * (xs[k] - slope * x[k] ** 2) + slope * bk
+            # Extend cross section to 0 assuming 1/v shape
+            if (k == 0 and a >= -4):
+                fb = fa.copy()
+                a = -y
+                self._calculate_f(fa, a)
+                h = fa - fb
+                sigma += xs[k] * x[k] * (y_inv_sq * h[1] + y_inv * h[0])
+            # Evaluate sigma*(y, T) from x[k] - y = 0 to 4
+            k = i
+            b = 0
+            self._calculate_f(fb, b)
+            while (b <= 4 and k < len(xs) - 1):
+                # Move to next point
+                fa = fb.copy()
+                k += 1
+                b = x[k] - y
+                # Calculate f and h functions
+                self._calculate_f(fb, b)
+                h = fa - fb
+                ak = y_inv_sq * h[2] + 2 * y_inv * h[1] + h[0]
+                bk = y_inv_sq * h[4] + 4 * y_inv * h[3] + 6 * h[2] + \
+                     4 * y * h[1] + y_sq * h[0]
+                slope = (xs[k] - xs[k - 1]) / (x[k] ** 2 - x[k - 1] ** 2)
+                sigma += ak * (xs[k] - slope * x[k] ** 2) + slope * bk
+            # Extend cross section to infinity assuming constant shape
+            if (k == len(xs) - 1 and b <= 4):
+                a = x[k] - y
+                self._calculate_f(fa, a)
+                sigma += xs[k] * (y_inv_sq * fa[2] + 2 * y_inv * fa[1] + fa[0])
+                
+            # Evaluate second term from x[k] + y = 0 to 4
+            if (y <= 4):
+                # swip signs on y
+                y = -y
+                y_inv = -y_inv
+                k = 0
+                # Calculate a and b based on 0 and x[0]
+                a = -y
+                b = x[k] - y
+                self._calculate_f(fa, a)
+                self._calculate_f(fb, b)
+                h = fa - fb
+                sigma = sigma - xs[k] * x[k] * (y_inv_sq * h[1] + y_inv * h[0])
+                while (b <= 4):
+                    fa = fb.copy()
+                    k += 1
+                    b = x[k] - y
+                    self._calculate_f(fb, b)
+                    h = fa - fb
+                    ak = y_inv_sq * h[2] + 2 * y_inv * h[1] + h[0]
+                    bk = y_inv_sq * h[4] + 4 * y_inv * h[3] + 6 * h[2] + \
+                         4 * y * h[1] + y_sq * h[0]
+                    slope = (xs[k] - xs[k - 1]) / (x[k] ** 2 - x[k - 1] ** 2)
+                    sigma = sigma - ak * (xs[k] - slope * x[k] ** 2) - \
+                            slope * bk
+            self.sigmaNew[i] = sigma
```

Comment:
Personally I think it would be better to just return the broadened cross section rather than add a new attribute